semantic_router/index/local.py

```diff
@@ -2,53 +2,68 @@
 from typing import List, Tuple, Optional
 from semantic_router.linear import similarity_matrix, top_scores
 from semantic_router.index.base import BaseIndex
-
+from semantic_router.route import Route
 
 class LocalIndex(BaseIndex):
     def __init__(self):
         super().__init__()
         self.type = "local"
         self.index: Optional[np.ndarray] = None
-        self.routes: Optional[np.ndarray] = None
+        self.routes: Optional[List[Route]] = None
+        self.route_names: Optional[List[str]] = None
         self.utterances: Optional[np.ndarray] = None
 
     class Config:  # Stop pydantic from complaining about  Optional[np.ndarray] type hints.
         arbitrary_types_allowed = True
 
     def add(
-        self, embeddings: List[List[float]], routes: List[str], utterances: List[str]
+        self, embeddings: List[List[float]], route: Route, utterances: List[str]
```

Comment:
we need to keep the methods the same across all of the index classes (that is why we use the `BaseIndex`, it should enforce a standard interface for all versions)

If we have all route utterances already in `route.utterances`, there's no need to pass `utterances` as a parameter here

Reply:
Hey James, thanks for the review. Good point I forgot to change the type hint in `BaseIndex`. I'm running `MyPy` and linting today, so hopefully that will catch any others of these types of mistake.

Also, great point about utterances will change that now.

Reply:
Done! Had to  make some other changes too, so that we can handle multiple routes being added at once.

See: https://github.com/aurelio-labs/semantic-router/pull/139/commits/58e78a6da3b4a3ef00840a9648bccf96a048b6d6